modules/plugins/languages/ruby.nix

```diff
@@ -0,0 +1,154 @@
+
+  cfg = config.vim.languages.ruby;
+
+  defaultServer = "rubyserver";
+  servers = {
+    rubyserver = {
+      package = pkgs.rubyPackages.solargraph;
+      lspConfig = ''
+        lspconfig.solargraph.setup {
+          capabilities = capabilities,
+          on_attach = attach_keymaps,
+          flags = {
+            debounce_text_changes = 150,
+          },
+          cmd = { "${pkgs.solargraph}/bin/solargraph", "stdio" }
+        }
+      '';
+    };
+  };
+
+  # testing
+
+  defaultFormat = "rubocop";
+  formats = {
+    rubocop = {
+      package = pkgs.rubyPackages.rubocop;
+      nullConfig = ''
+
+        local conditional = function(fn)
+          local utils = require("null-ls.utils").make_conditional_utils()
+          return fn(utils)
+        end
+
+        table.insert(
+          ls_sources,
+          null_ls.builtins.formatting.rubocop.with({
+            command="${pkgs.bundler}/bin/bundle",
+            args = vim.list_extend(
+              {"exec", "rubocop", "-a" },
+              null_ls.builtins.formatting.rubocop._opts.args
+            ),
+          })
+        )
+      '';
+    };
+  };
+
+  defaultDiagnosticsProvider = ["rubocop"];
+  diagnosticsProviders = {
+    rubocop = {
+      package = pkgs.rubyPackages.rubocop;
+      nullConfig = pkg: ''
+        table.insert(
+          ls_sources,
+          null_ls.builtins.diagnostics.rubocop.with({
+            command = "${lib.getExe pkg}",
+          })
+        )
+      '';
+    };
+  };
+in {
+  options.vim.languages.ruby = {
+    enable = mkEnableOption "Ruby language support";
+
+    treesitter = {
+      enable = mkEnableOption "Ruby treesitter" // {default = config.vim.languages.enableTreesitter;};
+      package = mkGrammarOption pkgs "ruby";
+    };
+
+    lsp = {
+      enable = mkEnableOption "Ruby LSP support" // {default = config.vim.languages.enableLSP;};
+
+      server = mkOption {
+        type = enum (attrNames servers);
+        default = defaultServer;
+        description = "Ruby LSP server to use";
+      };
+
+      package = mkOption {
+        description = "Ruby LSP server package, or the command to run as a list of strings";
+        type = either package (listOf str);
+        default = servers.${cfg.lsp.server}.package;
+        example = ''TODO'';
+      };
+    };
+
+    format = {
+      enable = mkEnableOption "Ruby formatter support" // {default = config.vim.languages.enableFormat;};
+
+      type = mkOption {
+        description = "Ruby formatter to use";
+        type = enum (attrNames formats);
+        default = defaultFormat;
+      };
+
+      package = mkOption {
+        description = "Ruby formatter package";
+        type = package;
+        default = formats.${cfg.format.type}.package;
+      };
+    };
+
+    extraDiagnostics = {
+      enable =
+        mkEnableOption "Ruby extra diagnostics support"
+        // {default = config.vim.languages.enableExtraDiagnostics;};
+
+      types = lib.nvim.types.diagnostics {
```

Comment:
You have an inherit from `lib.nvim.types`, it's fine to inherit `diagnostics` in the let in block and omit `lib.nvim.types.` here.